File: backend/src/main/java/com/back/domain/adoption/repository/AdoptionRepository.java
package com.back.domain.adoption.repository;

import com.back.domain.adoption.entity.Adoption;
import com.back.domain.member.entity.Member;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

public interface AdoptionRepository extends JpaRepository<Adoption, Long> {

    List<Adoption> findByMemberOrderByCreatedAtDesc(Member member);

    Optional<Adoption> findByIdAndMember(Long id, Member member);

    Optional<Adoption> findByIdAndPet_Member(Long id, Member member);

    List<Adoption> findByMember(Member member);

    List<Adoption> findByPet_MemberOrderByCreatedAtDesc(Member member);
} 

File: backend/src/main/java/com/back/domain/adoption/repository/AdoptionRepository.kt
package com.back.domain.adoption.repository

import com.back.domain.adoption.entity.Adoption
import com.back.domain.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository

interface AdoptionRepository : JpaRepository<Adoption, Long> {
    fun findByMemberOrderByCreatedAtDesc(member: Member): List<Adoption>

    fun findByIdAndMember(id: Long, member: Member): Adoption?

    // 언더스코어 주의 문제 다시 체킹
    fun findByIdAndPet_Member(id: Long, member: Member): Adoption?

    fun findByMember(member: Member): List<Adoption>

    // 언더스코어 주의 문제 다시 체킹
    fun findByPet_MemberOrderByCreatedAtDesc(member: Member): List<Adoption>
}